docs/extensions_preview.md

```diff
@@ -0,0 +1,45 @@
+---
+title: Extensions Preview
+layout: docs
+---
+
+Introducing Tilt Extensions! Extensions are a simple way to bundle up Tiltfile functionality that should be shared with the world.
+
+We think that there are lots of higher level patterns that people are implementing with Tilt that could benefit from sharing, socialization and standardization. For example many users have implemented their own ways to force all kubernetes resources in to a namespace, inject sidecars or run tests. Extensions provide a way for users to package up these ideas and share them with the community.
+
+Let's take a look at a simple example that shows why extensions are a powerful addition to Tilt. We'll follow it up with a more complicated example.
+
+## Simple Example: API Server Logs
+Depending on the kind of Kubernetes work you're doing it can be valuable to see the Kubernetes API Server logs. For example, if you were iterating on a Kubernetes controller that interacts heavily with the Kubernetes API. By default Tilt doesn't display those logs because they don't come from a resource that Tilt deployed. However, it's pretty easy to get tilt to display them using a `local_resource`:
+
+```python
+local_resource('API Server Logs', '', serve_cmd='kubectl logs -f -n kube-system kube-apiserver-docker-desktop')
+```
+
+This only works for Docker for Desktop on macOS, since it's hard coded to that pod name. But it would be easy to write a query to get the API server pod for any arbitrary Kubernetes cluster:
+
+```python
+#TODO(dmiller): is this right?
+api_server_pod_name = str(local('kubectl get pods --namespace kube-system -o=jsonpath="{.items..metadata.name}" -l component=kube-apiserver')).rstrip(\n)
+```
+
+Then you could compose them together in a function:
+
+```python
+def api_server_logs():
+  api_server_pod_name = str(local('kubectl get pods --namespace kube-system -o=jsonpath="{.items..metadata.name}" -l component=kube-apiserver')).rstrip(\n)
+  local_resource('API Server Logs', '', serve_cmd='kubectl logs -f -n %s' % api_server_pod_name )
+```
+
+That's it. You've created your first Tilt extension! Simply [open a PR](https://github.com/windmilleng/tilt-extensions/compare) against the official [Tilt Extensions Repo](https://github.com/windmilleng/tilt-extensions) to make this extension available to all Tilt users. All a user needs to do now is add a simple call to `load` to their Tiltfile that references the extension:
```

Comment:
ok, maybe this is making this explicitly a "write your first Tilt extension" doc

maybe we want one doc for writing and one for using? or just a TOC in this doc with sections "writing an extension", "finding/using extensions"

Reply:
Yeah I think I'll change this to writing an extension/why would you want to write an extension and then write a separate one for using extensions in a separate PR, since I think we should have some for realsies extensions to use before writing a usage guide 😅